NETDefault/UnitTestingWithNUnit&Moq/UnitTesting.md

```diff
@@ -0,0 +1,29 @@
+# Unit Testing with NUnit and Moq
+
+In this directory you will find a small project where we demonstrate the usage of NUnit and Moq to bring our code under test. Here, a _Service_ is being the subject under test, which depends on a _Gateway_ for completing the _GetDashboard_ call. Since we are unit testing, we do *not* want to include the external calls with the Gateway because that is not what we want to test and it would take far too long to complete. It is therefore that we create a fake instance of this gateway -  a.k.a. a mock, moq, substitute - for which we use [Moq](https://github.com/moq). Our _Service_ is the actual concrete class we want to test, which we do with [NUnit](https://github.com/moq).
+
+## Points of interest
```

Comment:
`TestCase` and `TestCaseSource` are too valuable to not also mention here.

Reply:
Added them in the description, and also two test examples.